Foundation/FileManager+Win32.swift

```diff
@@ -286,12 +287,12 @@ extension FileManager {
     internal func _canonicalizedPath(toFileAtPath path: String) throws -> String {
         var hFile: HANDLE = INVALID_HANDLE_VALUE
         path.withCString(encodedAs: UTF16.self) { link in
-            hFile = CreateFileW(link, GENERIC_READ, DWORD(FILE_SHARE_WRITE), nil, DWORD(OPEN_EXISTING), DWORD(FILE_FLAG_BACKUP_SEMANTICS), nil)
+            hFile = CreateFileW(link, GENERIC_READ, DWORD(FILE_SHARE_READ | FILE_SHARE_WRITE), nil, DWORD(OPEN_EXISTING), DWORD(FILE_FLAG_BACKUP_SEMANTICS), nil)
```

Comment:
- Please use `0` instead of `GENERIC_READ`
- Please use `FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE`
- Please add a comment to explain the need for the `FILE_FLAG_BACKUP_SEMANTICS` here.